contrib/opencensus-ext-azure/opencensus/ext/azure/metrics_exporter/heartbeat_metrics/heartbeat.py

```diff
@@ -48,17 +85,32 @@ def __init__(self):
             # Function apps
             self.properties[LabelKey("azfunction_appId", '')] = \
                 LabelValue(os.environ.get("WEBSITE_HOSTNAME"))
+        elif self.get_azure_compute_metadata():
+            # VM
+            if self.vm_data:
+                self.properties[LabelKey("azInst_vmId", '')] = \
+                    LabelValue(self.vm_data.get("vmId", ''))
+                self.properties[LabelKey("azInst_subscriptionId", '')] = \
+                    LabelValue(self.vm_data.get("subscriptionId", ''))
+                self.properties[LabelKey("azInst_osType", '')] = \
+                    LabelValue(self.vm_data.get("osType", ''))
 
-    def __call__(self):
-        """ Returns a derived gauge for the heartbeat metric.
+    def get_azure_compute_metadata(self):
+        try:
+            request_url = "{0}?{1}&{2}".format(
+                _AIMS_URI, _AIMS_API_VERSION, _AIMS_FORMAT)
+            response = requests.get(request_url, headers={"MetaData": "True"})
+        except requests.exceptions.ConnectionError:
+            # Not in VM
+            self.is_vm = False
```

Comment:
Is connection error enough to determine is not a VM?, maybe adding generic except as well to be sure doesn't think is a VM

Reply:
You are correct about the second except. I'll add it to catch anything else.